### ISSUE TITLE
Might be nice to parallelize PNG image loading

### ISSUE THREAD

**Issue by smcameron** [smcameron/space-nerds-in-space#331]
This would require separating the PNG loading/decoding step from the opengl texture binding stuff, since the opengl stuff can only be called from the main thread, and thus is serialized.  I believe most of the time loading images is actually spent in PNG decoding, so separating this into a first step in which all the PNG coding is done (in parallel) and a 2nd step where these loaded PNG images are made known to opengl (serially) should in theory result in significantly faster load times.

**Comment by smcameron** [comment]
Alright, I've committed the following:

8b8e542caa8af04e7d8b21508df9f19e2a507013 graph-dev: rename "deferred" texture loading functions
7f8d5b0160d68a8540cdca7984ef0dfd6a9ca925 graph-dev: fix tracking of loaded cubemap images
865f42f0812883d8e41e094c9e78739322daef7a graph-dev: delete old immediate texture loading functions
506e0d5af28e58e415c43fa8d40bb749228acbc2 stl-parser: use deferred texture loading functions
48f047466dcd1de679882473fe44a3c13d0a310f material: use deferred texture loading routines
c4c61ad093f4de9a449275466f448d10226b412f graph-dev: increase texture count limit
516d7b89930ca9d590bf4ffb0ae2a16444a86513 graph-dev: make texture reloading use work queues
f5a5f28af241be08767b8fe3f4bfd513c8a68dd8 mesh_viewer: used workqueue for decoding PNGs outside main render loop
6d65de9b6734ee3744ffc690f5310fb6957cfe31 graph_dev: set up texture loading work queues in graph_dev_setup()
3586fed45b50d18248f2b568ef03f5a87cc5d826 graph_dev: move send_new_textures_to_gpu out of snis_client
90edba75e3ee54de559b89c160f4f177967b1fa4 graph_dev: fix some data races
e5901d82438c043b40a5259c2b03103fca0d41a8 snis_client: use work queues to load cubemap textures
ec8f61902fd87342d217612d26fe80629d1d28cb snis_client: use workqueues to load simple textures to gpu
79a116da6f0a6f8379809e534b4f060251ec2a5d graph_dev: add functions to upload image data to GPU
eebd33e814360e9bbb2a9331a6a50843f463d978 graph_dev: add work queues for loading png images
f23ffba05ee9940e566409aecd05af15c9fa166a graph_dev: add checks to ensure textures loaded before attempting to render them
33bbf315b67a770cb9398d7a584b281a75f3b8d4 Add work queue system
1eb6c6d7956c9e1096dc62c44473f7decb0a0558 graph_dev_opengl: wrap glGenTextures() so we can track texture names
